File: easy/2053_kth_distinct_string_in_an_array/solution.py
from collections import Counter
from typing import List


class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        for s, count in Counter(arr).items():
            if count == 1:
                k -= 1

                if k == 0:
                    return s

        return ""

File: easy/2053_kth_distinct_string_in_an_array/solution.cpp
#include <string>
#include <unordered_map>
#include <vector>

class Solution {
public:
  std::string kthDistinct(std::vector<std::string> &arr, int k) {
    std::unordered_map<std::string, int> stringCounter;

    for (const std::string &s : arr)
      stringCounter[s]++;

    for (const std::string &s : arr) {
      if (stringCounter[s] == 1 && --k == 0)
        return s;
    }

    return "";
  }
};
